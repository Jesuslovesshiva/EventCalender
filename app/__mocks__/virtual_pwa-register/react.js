import {useState} from 'react'

const useRegisterSW = jest.fn(() => {
  const [needRefresh, setNeedRefresh] = useState(true)

  return {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker: () => {}
  }
})

// eslint-disable-next-line import/prefer-default-export
export {useRegisterSW}
