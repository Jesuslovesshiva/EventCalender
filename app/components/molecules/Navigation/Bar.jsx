import classNames from 'classnames'
import PropTypes from 'prop-types'

import NavigationItem from '@/components/atoms/NavigationItem'

import NavigationMenu from '../NavigationMenu'
import NavigationMenuItem from '../NavigationMenu/Item'
import NavigationSubMenu from '../NavigationMenu/SubMenu'

/* eslint-disable sort-keys */
const LEVELS = {
  ONE: 'one',
  TWO: 'two',
  THREE: 'three'
}
/* eslint-enable sort-keys */

const NavigationBar = props => {
  const className = classNames(
    'flex justify-center overflow-hidden order-last lg:order-2'
  )

  return (
    <nav
      className={classNames(
        'flex justify-center flex-1 flex-col lg:flex-row',
        'z-10',
        {
          'pt-1 bg-gradient-to-l bg-gradient-primary':
            props.level === LEVELS.ONE,
          'pt-[3px] bg-gray-light': props.level === LEVELS.TWO,
          'py-[3px] bg-white': props.level === LEVELS.THREE
        }
      )}
    >
      {props.cta}
      <div className={className}>
        {props.items.map(item => {
          if (item.children?.length) {
            return (
              <NavigationMenu
                key={item.label}
                menuButton={({
                  isOpen,
                  onKeyDown,
                  onMouseEnter,
                  onMouseLeave,
                  ref
                }) => (
                  <NavigationItem
                    hasSublevel={props.level !== LEVELS.ONE}
                    isDisabled={item.folder}
                    isOpen={isOpen}
                    level={props.level}
                    onKeyDown={onKeyDown}
                    onMouseEnter={onMouseEnter}
                    onMouseLeave={onMouseLeave}
                    ref={ref}
                    to={`${props.basePath}${item.path}`}
                  >
                    {item.label}
                  </NavigationItem>
                )}
              >
                {item.children.map(child => {
                  if (child.children?.length) {
                    return (
                      <NavigationSubMenu
                        key={child.label}
                        to={`${props.basePath}${item.path}/${child.path}`}
                        isLabelDisabled={child.folder}
                        label={child.label}
                      >
                        {child.children.map(grandchild => (
                          <NavigationMenuItem
                            isDisabled={grandchild.folder}
                            key={grandchild.label}
                            to={`${props.basePath}${item.path}/${child.path}/${grandchild.path}`}
                          >
                            {grandchild.label}
                          </NavigationMenuItem>
                        ))}
                      </NavigationSubMenu>
                    )
                  }
                  return (
                    <NavigationMenuItem
                      isDisabled={child.folder}
                      key={child.label}
                      to={`${props.basePath}${item.path}/${child.path}`}
                    >
                      {child.label}
                    </NavigationMenuItem>
                  )
                })}
              </NavigationMenu>
            )
          }

          return (
            <NavigationItem
              key={item.label}
              level={props.level}
              to={`${props.basePath}${item.path}`}
            >
              {item.label}
            </NavigationItem>
          )
        })}
      </div>
      {props.settings}
    </nav>
  )
}

NavigationBar.propTypes = {
  basePath: PropTypes.string,
  cta: PropTypes.node,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      children: PropTypes.array,
      folder: PropTypes.bool,
      label: PropTypes.string,
      path: PropTypes.string
    })
  ),
  level: PropTypes.oneOf(Object.values(LEVELS)),
  settings: PropTypes.node
}

NavigationBar.defaultProps = {
  basePath: '/',
  cta: null,
  items: [],
  level: LEVELS.ONE,
  settings: null
}

export {LEVELS}
export default NavigationBar
