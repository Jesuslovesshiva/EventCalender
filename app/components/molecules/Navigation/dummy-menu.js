/* eslint-disable max-lines,sort-keys */
export default [
  {
    label: 'Trainingsplan',
    path: 'training-plan'
  },
  {
    label: 'Training',
    path: 'training',
    children: [
      {
        label: 'Continuous Training',
        path: 'continuous'
      },
      {
        label: 'Fartlek Training',
        path: 'fartlek'
      },
      {
        label: 'Circuit Training',
        path: 'circiut'
      },
      {
        label: 'Interval Training',
        path: 'interval',
        children: [
          {
            label: 'High Intensity',
            path: 'high-intensity'
          },
          {
            label: 'Medium Intensity',
            path: 'medium-intensity'
          },
          {
            label: 'Low Intensity',
            path: 'low-intensity'
          }
        ]
      },
      {
        label: 'Flexibility/Mobility Training',
        path: 'flexibility-mobility'
      },
      {
        label: 'Weight Training',
        path: 'weight'
      },
      {
        label: 'Plyometric Training',
        path: 'plyometric'
      },
      {
        label: 'Speed/Agility/Quickness',
        path: 'speed-agility-quickness'
      }
    ]
  },
  {
    label: 'Games',
    path: 'games',
    children: [
      {
        label: 'Regional League',
        path: 'regional-league'
      },
      {
        label: 'International League',
        path: 'international-league'
      },
      {
        label: 'Friendly games',
        path: 'friendly-games'
      }
    ]
  },
  {
    label: 'Medicine',
    path: 'medicine',
    children: [
      {
        label: 'Injuries',
        path: 'injuries'
      },
      {
        label: 'Treatments',
        path: 'treatments'
      },
      {
        label: 'Analysis',
        path: 'analysis',
        children: [
          {
            label: 'Body Region',
            path: 'body-region'
          },
          {
            label: 'Type of Injury',
            path: 'type-of-injury'
          },
          {
            label: 'Injury Susceptibility',
            path: 'injury-susceptibility'
          },
          {
            label: 'Season Comparison',
            path: 'season-comparison'
          },
          {
            label: 'Availability',
            path: 'availability'
          },
          {
            label: 'Missed Games',
            path: 'missed-games'
          }
        ]
      }
    ]
  },
  {
    label: 'Diagnostics',
    path: 'diagnostics',
    children: [
      {
        label: 'Speed',
        path: 'speed'
      },
      {
        label: 'Endurance',
        path: 'endurance'
      },
      {
        label: 'Body Measurements',
        path: 'body-measurements'
      },
      {
        label: 'Jumping',
        path: 'jumping'
      }
    ]
  },
  {
    label: 'Team',
    path: 'team',
    children: [
      {
        label: 'Top-Players',
        path: 'top-players'
      },
      {
        label: 'Team Assignment',
        path: 'team-assignment'
      },
      {
        label: 'Player Evaluations',
        path: 'player-evaluations'
      }
    ]
  },
  {
    label: 'Philosophy',
    path: 'philosophy',
    children: [
      {
        label: 'About the Team',
        path: 'about-the-team',
        children: [
          {
            label: 'About the Players',
            path: 'players'
          },
          {
            label: 'Achievements',
            path: 'achievements'
          },
          {
            label: 'Goals',
            path: 'goals'
          }
        ]
      }
    ]
  },
  {
    label: 'Administration',
    path: 'administration',
    children: [
      {
        label: 'Players',
        path: 'players'
      },
      {
        label: 'Queries',
        path: 'queries',
        children: [
          {
            label: 'Players',
            path: 'players'
          },
          {
            label: 'Employees',
            path: 'employees'
          }
        ]
      },
      {
        label: 'Task & Objectives',
        path: 'task-objectives',
        children: [
          {
            label: 'Tasks',
            path: 'tasks'
          },
          {
            label: 'Objectives',
            path: 'objectives'
          }
        ]
      },
      {
        label: 'Admin',
        path: 'admin',
        children: [
          {
            label: 'Panel',
            path: 'panel'
          },
          {
            label: 'Analytics',
            path: 'analytics'
          }
        ]
      }
    ]
  }
]
