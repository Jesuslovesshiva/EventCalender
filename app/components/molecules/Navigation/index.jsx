import {useRouter} from '@tanstack/react-location'
import PropTypes from 'prop-types'
import React, {useMemo} from 'react'

import NavigationBar, {LEVELS} from './Bar'
import MENU from './dummy-menu'

const findSubMenu = (menu, path) =>
  menu?.find(item => item.path === path) ?? null

const findActiveSubMenu = (menu, path, subPath) => {
  const subMenu = findSubMenu(menu, path)

  if (findSubMenu(subMenu?.children, subPath)) {
    return subMenu.children
  }

  return null
}

const Navigation = props => {
  const router = useRouter()
  const [firstLevelPath, secondLevelPath, thirdLevelPath] = useMemo(
    () => router.state.location.pathname.split('/').filter(Boolean),
    [router.state.location]
  )
  const secondLevelMenu = useMemo(
    () => findActiveSubMenu(MENU, firstLevelPath, secondLevelPath),
    [firstLevelPath, secondLevelPath]
  )
  const thirdLevelMenu = useMemo(
    () => findActiveSubMenu(secondLevelMenu, secondLevelPath, thirdLevelPath),
    [secondLevelPath, secondLevelMenu, thirdLevelPath]
  )

  return (
    <>
      <NavigationBar
        cta={props.cta}
        items={MENU}
        level={LEVELS.ONE}
        settings={props.settings}
      />
      {Boolean(secondLevelMenu) && (
        <NavigationBar
          basePath={`/${firstLevelPath}/`}
          items={secondLevelMenu}
          level={LEVELS.TWO}
        />
      )}
      {Boolean(thirdLevelMenu) && (
        <NavigationBar
          basePath={`/${firstLevelPath}/${secondLevelPath}/`}
          items={thirdLevelMenu}
          level={LEVELS.THREE}
        />
      )}
    </>
  )
}

Navigation.propTypes = {
  cta: PropTypes.node,
  settings: PropTypes.node
}

Navigation.defaultProps = {
  cta: null,
  settings: null
}

export default React.memo(Navigation)
