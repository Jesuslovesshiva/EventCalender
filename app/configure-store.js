import {
  combineReducers,
  configureStore as configureToolkitStore,
  isPlain
} from '@reduxjs/toolkit'
import {createBrowserHistory} from '@tanstack/react-location'
import {createReduxHistoryContext} from 'redux-first-history'
import {createInjectorsEnhancer} from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import ActionSubscription, {
  actionListenerMiddleware
} from '@/components/organisms/ActionSubscription'

const actionEmitter = new ActionSubscription.ActionEmitter()

const {createReduxHistory, routerMiddleware, routerReducer} =
  createReduxHistoryContext({
    history: createBrowserHistory()
  })

const sagaMiddleware = createSagaMiddleware()

const configureStore = () => {
  const {run: runSaga} = sagaMiddleware
  const middlewares = [
    sagaMiddleware,
    routerMiddleware,
    actionListenerMiddleware(actionEmitter)
  ]

  const createReducer = injectedReducers =>
    combineReducers({
      router: routerReducer,
      ...injectedReducers
    })
  const enhancers = [createInjectorsEnhancer({createReducer, runSaga})]

  const isSerializable = value =>
    value instanceof Error ||
    value instanceof File ||
    value instanceof Response ||
    isPlain(value)

  const store = configureToolkitStore({
    // eslint-disable-next-line no-undef
    devTools: process.env.NODE_ENV !== 'production',
    enhancers,
    middleware: getDefaultMiddleware => [
      ...getDefaultMiddleware({serializableCheck: {isSerializable}}),
      ...middlewares
    ],
    reducer: createReducer()
  })

  store.injectedReducers = {}
  store.injectedSagas = {}

  return store
}

const store = configureStore()
const history = createReduxHistory(store)

export {configureStore, history, actionEmitter}

export default store
