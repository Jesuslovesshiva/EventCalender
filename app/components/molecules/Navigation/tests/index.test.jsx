import {act, waitFor} from '@testing-library/react'

import {history} from '@/configure-store'
import render from '@/utils/test-utils/custom-render'

import Navigation from '../index'

const options = {wrapperProps: {hasRouter: true}}
const renderComponent = () => render(<Navigation />, options)

describe('Navigation', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render one navigation level', async () => {
    const {container} = renderComponent()
    act(() => {
      history.push('/medicine')
    })
    await waitFor(() =>
      expect(container.querySelectorAll('nav')).toHaveLength(1)
    )
  })

  it('should render two navigation levels', async () => {
    const {container} = renderComponent()
    act(() => {
      history.push('/medicine/analysis')
    })
    await waitFor(() =>
      expect(container.querySelectorAll('nav')).toHaveLength(2)
    )
  })

  it('should render three navigation levels', async () => {
    const {container} = renderComponent()
    act(() => {
      history.push('/medicine/analysis/missed-games')
    })
    await waitFor(() =>
      expect(container.querySelectorAll('nav')).toHaveLength(3)
    )
  })

  it('should not render 2nd level menu on unknown path', () => {
    const {container} = renderComponent()
    act(() => {
      history.push('/medicine/foo')
    })
    expect(container.querySelectorAll('nav')).toHaveLength(1)
  })

  it('should not render 3rd level menu on unknown path', () => {
    const {container} = renderComponent()
    act(() => {
      history.push('/medicine/analysis/bar')
    })
    expect(container.querySelectorAll('nav')).toHaveLength(2)
  })
})
