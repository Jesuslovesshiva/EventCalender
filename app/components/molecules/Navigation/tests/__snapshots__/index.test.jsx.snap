// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
<nav
  class="flex justify-center flex-1 flex-col lg:flex-row z-10 pt-1 bg-gradient-to-l bg-gradient-primary"
>
  <div
    class="flex justify-center overflow-hidden order-last lg:order-2"
  >
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/training-plan"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Trainingsplan
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/training"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Training
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/games"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Games
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/medicine"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Medicine
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/diagnostics"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Diagnostics
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/team"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Team
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/philosophy"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Philosophy
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/asa"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        ASA
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
    <a
      class="font-bold inline-flex flex-col transition motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-inset text-[13px] leading-[15px] text-white px-3 pt-2.5 pb-[11px] rounded-t focus-visible:ring-white hover:text-primary hover:bg-white hover:shadow"
      href="/administration"
      tabindex="0"
    >
      <div
        class="flex items-center"
      >
        Administration
      </div>
      <div
        class="relative"
      >
        <div
          class="w-full absolute top h-[2px] left-0 right-0 top-0 rounded-full bg-current motion-reduce:transition-none transition-[opacity,transform] mt-px opacity-0 translate-y-0.5"
        />
      </div>
    </a>
  </div>
</nav>
`;
